simapp/app.go

```diff
@@ -357,7 +358,10 @@ func NewSimApp(
 		upgradetypes.ModuleName, minttypes.ModuleName, distrtypes.ModuleName, slashingtypes.ModuleName,
 		evidencetypes.ModuleName, stakingtypes.ModuleName, ibchost.ModuleName,
 	)
-	app.mm.SetOrderEndBlockers(crisistypes.ModuleName, govtypes.ModuleName, stakingtypes.ModuleName)
+
+	// NOTE: slashing module endblocker should run before staking module since MsgUnjail epoch action should run before making
+	// validator set update on staking module
+	app.mm.SetOrderEndBlockers(crisistypes.ModuleName, govtypes.ModuleName, slashingtypes.ModuleName, stakingtypes.ModuleName)
```

Comment:
Is there a clean way we can make the app fail if slashing's end blocker is ran after staking's end blocker?
Fairly worried people are going to get this wrong.

Reply:
One way would be slashing endblocker to update flag on staking keeper and check it on staking endblocker before running and output warning or panic. (But it could be an ugly way)